### PR TITLE
inline FluentIterable.of to be compatible with guava <18

### PR DESCRIPTION
**Pull request by Xenorith**

### CODE REVIEW

**Comment by pwoody** [comment]
Can we shade guava?

**Comment by markelliot** [comment]
I'd rather not, would prefer to rewrite this to remove the Guava dep rather than shade just for this one liner.

**Comment by pwoody** [comment]
cool - just something to be wary of for all of our libraries that interact with spark/hadoop. guava versions always cause issues (hence spark shaded theirs).

**Comment by markelliot** [comment]
We use so little of the Guava library that forcing it to whatever version you need is probably fine in the current mode. If we force shading of the Guava Optional class then that sort of ruins how this gets used, so I'm pretty hesitant to do that.

**Comment by uschi2000** [comment]
any objections to merging this for a 0.7.1 release?

File: http-clients/src/main/java/com/palantir/remoting/http/GuavaOptionalAwareContract.java [hunk 1]
```diff
@@ -67,7 +68,7 @@ public GuavaOptionalAwareContract(Contract delegate) {
                     Class<?> cls = parameterTypes[i];
                     if (cls.equals(Optional.class)) {
                         FluentIterable<Class<?>> paramAnnotations =
-                                FluentIterable.of(annotations[i]).transform(EXTRACT_CLASS);
+                                FluentIterable.from(Lists.newArrayList(annotations[i])).transform(EXTRACT_CLASS);
```
Comment (pnepywoda) [inline http-clients/src/main/java/com/palantir/remoting/http/GuavaOptionalAwareContract.java:71]: ImmutableList.of(). whatever. do it next time

**Comment by pnepywoda** [comment]
+1
We may want to actually downgrade guava in our widely-used libraries to guava 16, or even 14 which is what spark uses (http://mvnrepository.com/artifact/org.apache.spark/spark-core_2.10/1.6.0)

File: http-clients/src/main/java/com/palantir/remoting/http/GuavaOptionalAwareContract.java (hunk 1 above)
Comment (Xenorith) [inline http-clients/src/main/java/com/palantir/remoting/http/GuavaOptionalAwareContract.java:71]: went with inlining exactly what was in guava

  public static <E> FluentIterable<E> of(E[] elements) {
    return from(Lists.newArrayList(elements));
  }

**Comment by uschi2000** [comment]
can you rebase please, @Xenorith ?

**Pull request merged**